ctapipe/reco/sklearn.py

```diff
@@ -507,6 +501,325 @@ def predict_table(self, key, table: Table) -> Table:
+        allow_none=True,
+        help="Which scikit-learn regression model to use.",
+    ).tag(config=True)
+
+    sign_config = Dict({}, help="kwargs for the sklearn classifier").tag(config=True)
+    sign_cls = Enum(
+        SUPPORTED_CLASSIFIERS.keys(),
+        default_value=None,
+        allow_none=True,
+        help="Which scikit-learn classification model to use.",
+    ).tag(config=True)
+
+    stereo_combiner_cls = ComponentName(
+        StereoCombiner,
+        default_value="StereoMeanCombiner",
+        help="Which stereo combination method to use",
+    ).tag(config=True)
+
+    load_path = Path(
+        default_value=None,
+        allow_none=True,
+        help="If given, load serialized model from this path",
+    ).tag(config=True)
+
+    def __init__(self, subarray=None, norm_models=None, sign_models=None, **kwargs):
+        # Run the Component __init__ first to handle the configuration
+        # and make `self.load_path` available
+        Component.__init__(self, **kwargs)
+
+        if self.load_path is None:
+            if self.norm_cls is None or self.sign_cls is None:
+                raise TraitError(
+                    "Must provide `norm_cls` and `sign_cls` if not loading from file"
+                )
+
+            if subarray is None:
+                raise TypeError(
+                    "__init__() missing 1 required positional argument: 'subarray'"
+                )
+
+            super().__init__(subarray, **kwargs)
+            self.quality_query = MLQualityQuery(parent=self)
+            self.feature_generator = FeatureGenerator(parent=self)
+
+            # to verify settings
+            self._new_models()
+
+            self._norm_models = {} if norm_models is None else norm_models
+            self._sign_models = {} if sign_models is None else sign_models
+            self.unit = None
+            self.stereo_combiner = StereoCombiner.from_name(
+                self.stereo_combiner_cls,
+                prefix=self.prefix,
+                property=self.property,
+                parent=self,
+            )
+        else:
+            loaded = self.read(self.load_path)
+            if (
+                subarray is not None
+                and loaded.subarray.telescope_types != subarray.telescope_types
+            ):
+                self.log.warning(
+                    "Supplied subarray has different telescopes than subarray loaded from file"
+                )
+            self.__dict__.update(loaded.__dict__)
+            self.subarray = subarray
+
+    def _new_models(self):
+        norm_regressor = SUPPORTED_REGRESSORS[self.norm_cls](**self.norm_config)
+        sign_classifier = SUPPORTED_CLASSIFIERS[self.sign_cls](**self.sign_config)
+        return norm_regressor, sign_classifier
+
+    def _table_to_y(self, table, mask=None):
+        """
+        Extract target values as numpy array from input table
+        """
+        # make sure we use the unit that was used during training
```

Comment:
this could be simplified to:

```
if self.unit is not None:
    norm = table[mask][self.target].quantity.to_value(self.unit)
else:
    norm = np.array(table[self.target][mask])

abs_norm = np.abs(norm)
sign_norm = np.sign(norm)

if self.log_target:
    # check < 0
    abs_norm = np.log(abs_norm)

return abs_norm, sign_norm
```

Reply:
Good point, either we move it above so we get the nice error or we remove it and just assume this cannot happen

Reply:
I think removing it here is fine, since this should never fail if a correct `mask` is passed.